src/System.ValueTuple/src/System/ValueTuple/ValueTuple.cs

```diff
@@ -338,6 +338,7 @@ bool IStructuralEquatable.Equals(object other, IEqualityComparer comparer)
 
             var objTuple = (ValueTuple<T1>)other;
 
+            if (comparer == null) comparer = EqualityComparer<object>.Default;
             return comparer.Equals(Item1, objTuple.Item1);
```

Comment:
I assume that shortcutting https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Collections/Generic/EqualityComparer.cs#L208-L216 where `T` is `object` with `object.Equals(object, object)` is the way to go to avoid the lookup and everything?